core/src/main/java/org/apache/carbondata/core/scan/result/vector/impl/CarbonColumnVectorImpl.java

```diff
@@ -367,7 +378,14 @@ public void setBlockDataType(DataType blockDataType) {
   }
 
   @Override public void setLazyPage(LazyPageLoader lazyPage) {
-    lazyPage.loadPage();
+    this.lazyPage = lazyPage;
+  }
+
+  public void loadPage() {
+    if (lazyPage != null) {
+      lazyPage.loadPage();
+    }
+    loaded = true;
```

Comment:
if lazyPath is null, still set to true?

Reply:
yes, it is needed, for first call it sets to true so that next call will not come here.